### ISSUE TITLE
False positive on relative imports (not supported)

### ISSUE THREAD

**Issue by andersfischernielsen** [Arkq/flake8-requirements#1]
Hi, great plugin! 

I'm currently writing a static analysis tool for validating Python code in the [Robot Operating System (ROS)](http://www.ros.org) where I'm using `flake8` and `flake8-requirements` to check for certain import errors.

Unfortunately I am getting false positives for relative imports, more specifically I have the following structure:

    src
    ├─ file_with_false_positives.py
    └─ my_module
      └─ my_module.py

The import in `file_with_false_positives.py` would be:
    
    from my_module import * 

and I am getting an error that `I900 'module_folder' not listed as a requirement`. 

It seems that `flake8-requirements` does not support registering relative imports. Is that true? 
If so, would it be possible to add relative import resolution to `flake8-requirements`?

**Comment by andersfischernielsen** [comment]
Could the relative imports be resolved using

```python 
import pkgutil
search_path = ['.']
all_modules = [x[1] for x in pkgutil.iter_modules(path=search_path)]
print(all_modules)
```

for a given file?

In the example above the code above spits out `my_module`.

**Comment by Arkq** [comment]
I think, that the problem is that you are not using relative import in the
full meaning of that word. You are using absolute import, however python
can resolve it because the search path contains current directory (which is
added by Python on startup). The proper way of using relative import is to
have a real module (with __init__.py file) and import it with a dot
notation.

See this: https://stackoverflow.com/a/14132912

Working example:

test/
  |-- mymodule/
  |      |-- __init__.py
  |      |-- tools.py
  |-- __init__.py
  |-- main.py

$ cat test/main.py
from .mymodule import tools
tools.test()
$ cat test/mymodule/tools.py
def test():
    print("Hello Module!")
$ python -m test.main
Hello Module!

The purpose of this checker is to test whether created *package* has valid
requirements. Which implies, that you have a correct Python package :)
Anyway, using absolute imports for a "relative" import might be
catastrophic. Imagine that your "my_module" is called e.g. "logging" - from
logging import xxx. Using it in that way, forces you to worry about name
collisions, which is not good in the terms of longevity and portability -
some future versions of Python might have new standard module called
"my_module".

>

**Comment by andersfischernielsen** [comment]
I'll might try to see if I can integrate some kind of __init__.py auto-generation. 

Since I'm simply working on other developers' existing codebases (with known bugs we want to detect) I can't change the source too much. 

Thanks for the in-depth answer anyways! 👍
